src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java

```diff
@@ -603,46 +571,44 @@ protected final boolean passesCallThreshold(final double conf) {
-            final double log10ACeq0Likelihood = log10GenotypeLikelihoods[0];
-            final double log10ACeq0Prior = log10Priors[0];
-            final double log10ACeq0Posterior = log10ACeq0Likelihood + log10ACeq0Prior;
-
-            // If the Maximum a-posteriori AC is 0 then the profile value must be 0.0 as per existing code; it does
-            // not matter whether a AC > 0 is at all plausible.
-            boolean mapACeq0 = true;
-            for (int AC = 1; AC < log10Priors.length; AC++) {
-                if (log10Priors[AC] + log10GenotypeLikelihoods[AC] > log10ACeq0Posterior) {
-                    mapACeq0 = false;
-                    break;
-                }
-            }
-            if (mapACeq0) {
-                return 0.0;
-            }
-
-            //TODO bad way to calculate AC > 0 posterior that follows the current behaviour of ExactAFCalculator (StateTracker)
-            //TODO this is the lousy part... this code just adds up lks and priors of AC != 0 before as if
-            //TODO Sum(a_i * b_i) is equivalent to Sum(a_i) * Sum(b_i)
-            //TODO This has to be changed not just here but also in the AFCalculators (StateTracker).
-            final double log10ACgt0Likelihood = MathUtils.approximateLog10SumLog10(log10GenotypeLikelihoods, 1, log10GenotypeLikelihoods.length);
-            final double log10ACgt0Prior = MathUtils.approximateLog10SumLog10(log10Priors, 1, log10Priors.length);
-            final double log10ACgt0Posterior = log10ACgt0Likelihood + log10ACgt0Prior;
-            final double log10PosteriorNormalizationConstant = MathUtils.approximateLog10SumLog10(log10ACeq0Posterior, log10ACgt0Posterior);
-            //TODO End of lousy part.
-
-            final double normalizedLog10ACeq0Posterior = log10ACeq0Posterior - log10PosteriorNormalizationConstant;
-            // This is another condition to return a 0.0 also present in AFCalculator code as well.
-            if (normalizedLog10ACeq0Posterior >= QualityUtils.qualToErrorProbLog10(configuration.genotypeArgs.STANDARD_CONFIDENCE_FOR_EMITTING)) {
-                return 0.0;
+        Utils.nonNull(log10GenotypeLikelihoods, "the input likelihoods cannot be null");
+        Utils.validateArg(log10GenotypeLikelihoods.length == this.configuration.genotypeArgs.samplePloidy + 1, "wrong likelihoods dimensions");
+
+        final double[] log10Priors = log10AlleleFrequencyPriorsSNPs.forTotalPloidy(this.configuration.genotypeArgs.samplePloidy);
+        final double log10ACeq0Likelihood = log10GenotypeLikelihoods[0];
+        final double log10ACeq0Prior = log10Priors[0];
+        final double log10ACeq0Posterior = log10ACeq0Likelihood + log10ACeq0Prior;
+
+        // If the maximum a posteriori AC is 0 then the profile value must be 0.0 as per existing code; it does
+        // not matter whether AC > 0 is at all plausible.
```

Comment:
This diff makes me do a double-take every time, but I actually didn't touch the method.  The reason it shows up is that it was originall:
```java
if (log10GenotypeLikelihoods == null) {
  throw new IllegalArgumentException("the input likelihoods cannot be null");
} else if (log10GenotypeLikelihoods.length != this.configuration.genotypeArgs.samplePloidy + 1) {
  throw new IllegalArgumentException("wrong likelihoods dimensions");
 } else {
   method goes here. . .
```